pkg/controller/cloneset/cloneset_controller.go

```diff
@@ -278,6 +305,20 @@ func (r *ReconcileCloneSet) doReconcile(request reconcile.Request) (res reconcil
 	}
 	*newStatus.CollisionCount = collisionCount
 
+	if !isPreDownloadDisabled {
+		if currentRevision.Name != updateRevision.Name {
+			// pre-download images for new revision
+			if err := r.createImagePullJobsForInPlaceUpdate(instance, currentRevision, updateRevision); err != nil {
+				klog.Errorf("Failed to create ImagePullJobs for %s: %v", request, err)
+			}
+		} else {
+			// delete ImagePullJobs if revisions have been consistent
+			if err := r.deleteImagePullJobs(instance); err != nil {
+				klog.Errorf("Failed to delete imagepulljobs for %s: %v", request, err)
```

Comment:
what if the cloneset rollout partially, and rollback, in such case, new imagejob will not be recycled

Reply:
Even if the cloneset rollback, the imagepulljob will still download image on all nodes. We have not to create it again.